### PR TITLE
Dev/twegner/issue 16

### PR DESCRIPTION
**Pull request by trwegner**
Removing MessageEvent to address #16.  The Event class already implements the requested extensibility as I understand the linked issues/threads.  While resolving this issue I found that Java implmentation uses SpanContext.Blank so updated code here to reflect that as well.  Warnings for touched files were fixed too.

### CODE REVIEW

File: src/OpenTelemetry.Abstractions/Trace/Export/SpanData.cs [hunk 1]
```diff
@@ -118,25 +112,24 @@ public sealed class SpanData : ISpanData
...
-            if (messageEvents == null)
+            if (events == null)
             {
-                messageEvents = TimedEvents<IMessageEvent>.Create(new List<ITimedEvent<IMessageEvent>>(), 0);
+                events = TimedEvents<IEvent>.Create(new List<ITimedEvent<IEvent>>(), 0);
             }
 
-            var messageEventsList = new List<ITimedEvent<IMessageEvent>>();
-            foreach (ITimedEvent<IMessageEvent> timedEvent in messageEvents.Events)
+            var eventsList = new List<ITimedEvent<IEvent>>();
```
Comment (lmolkova) [inline src/OpenTelemetry.Abstractions/Trace/Export/SpanData.cs]: it doesn't seem like eventsList or timedMessageEvents are used anywhere. perhaps there was some bug here?

File: src/OpenTelemetry.Abstractions/Trace/Event.cs [hunk 2]
```diff
@@ -55,7 +55,7 @@ public static IEvent Create(string name)
         /// Returns a new <see cref="Event"/> with the provided name and set of attributes.
         /// </summary>
         /// <param name="name">The text name for the <see cref="Event"/>.</param>
-        /// <param name="attributes">The <see cref="IDictionary{string, IAttributeValue}"/> of attributes for the <see cref="Event"/>.</param>
+        /// <param name="attributes">The <see cref="IDictionary{String, IAttributeValue}"/> of attributes for the <see cref="Event"/>.</param>
```
Comment (lmolkova) [inline src/OpenTelemetry.Abstractions/Trace/Event.cs]: nit: why `String` is needed?

Comment (SergeyKanzhelev) [reply src/OpenTelemetry.Abstractions/Trace/Event.cs]: I believe it was giving a warning otherwise as doc parser didn't find the type =)

Comment (trwegner) [reply src/OpenTelemetry.Abstractions/Trace/Event.cs]: The change is for **Warning CS1658 Type parameter declaration must be an identifier not a type. See also error CS0081**.

Comment (lmolkova) [reply src/OpenTelemetry.Abstractions/Trace/Event.cs]: got it, thanks for the explanation!

**Comment by SergeyKanzhelev** [comment]
@trwegner let me know if you want to look at what @lmolkova found with un-used variable. Or want me to merge now.

**Comment by trwegner** [comment]
Let's go ahead an merge.  I'll take a look and open a new issue to address.

**Pull request merged**